alias-forwarder-email-service/README.md

```diff
@@ -0,0 +1,113 @@
+
+This proposal merges the concepts of:
+-
+- having a dedicated email address for each customer
+- using the `+` sign trick to correctly redirect requests
+
+Let's consider customer `shiba`. We can create a dedicated email address `shiba@giantswarm.io` and then use the `deliveredto` Gmail filter to filter emails out. Some examples could be:
+
+- `deliveredto:"shiba+urgent+aws@giantswarm.io"`
+- `deliveredto:"shiba+support+apps@giantswarm.io"` (or *packs* :) )
+
+#### Pros
+
+- when a customer joins/leaves we can simply create/remove the email address
+- "low" price? One inbox per customer
+- no alias logic involved: this is a real email address
+
+#### Cons
+
+- we have to manage the different combinations and one inbox per customer
+
+### Proposal #2: just two email addresses
+
+This proposal relies on using the `+` sign trick to correctly redirect requests.
+
+We can define two email addresses: `urgent@giantswarm.io` and `support@giantswarm.io`.
+
+Let's consider customer `shiba`. We can discern between their requests depending on the additional information present after the first `+` sign. Some examples could be:
+
+- `deliveredto:"urgent+shiba+aws@giantswarm.io"`
+- `deliveredto:"support+shiba+apps@giantswarm.io"` (or *packs* :) )
+
+#### Pros
+
+- no additional inboxes with respect to now
+- no alias logic involved: these are a real email addresses
+
+#### Cons
+
+- we have to manage all the different combinations in the same inbox
+- when a customer leaves, we may need to go and modify some rules rather than simply deleting an inbox
+  - admittedly, we think and hope there won't be that many customers leaving us
+
+## Questions
+
+The following questions arise for both proposals:
+
+- having the customer in the email is not really useful at the moment. Should we still go for it "just in case"?
+  - it doesn't hurt
+- can we choose the action to be taken (forward to slack/opsgenie) depending on the `deliveredto` address?
+  - YES, we can!
+- can the various processes be automated somehow? Maybe using a [Google Apps Script](https://script.google.com/)?
+  - (scenario 1) email address creation/deletion
+  - (both scenarios) "routes": support to slack vs. urgent to opsgenie; different areas
+  I haven't been able to find a sensible solution so far, but we can dig into this a bit more.
+- can we use wildcards in Gmail filters?
+  - no, we can't
+# Final decisions
+
+Considering the pros and cons of each solution and the number of comments received, I believe we can proceed as follows:
+
+- Only use the support@giantswarm.io and urgent@giantswarm.io email addresses
+- With `support`, we only specify the customer name, so that the message is forwarded to the customer's Slack channel: `support+customer@giantswarm.io`
+- With `urgent`, we only specify the area, so that the message is forwarded to the correct area in Opsgenie: `urgent+area@giantswarm.io`. The areas are the following:
+
+  Cloud providers:
```

Comment:
I have some concerns about the selection:
It seems to me like this could cause confusion for customers. What would they do if they can't access the k8s API on AWS? set `aws`, `connectivity` or both? I feel like this could lead very easily to customers to always select their cloud provider and never a cloud native pack because they lack the knowledge to determine what the issue is.

Reply:
@MarcelMue this is a good point. I would say that the distinction, although not really black-white, is that the cloud native packs quite strictly involve applications. An API not reachable during the night seems to me more of an infrastructure problem rather than a problem with some application regarding connectivity (I imagine Kong as part of that pack, for example).

Which categories would you suggest? 

By the way, I recognize that this is a complex problem, but I'd rather start with a first version and improve iteratively than remain stuck trying to get the "perfect" combination. I think this is still an improvement with respect to the current configuration (and I may be wrong 😄).

Reply:
One idea would be to have the actual names of technologies instead of the packs. The problem would the amount of combinations then. 
e.g. instead of `connectivity` there is `nginx`, `kong`, `coredns` ...
And `devex` is then `flux` and so on

I just currently feel like a developer / ops person might not have the overview in a stressful situation which pack includes their application / issue. I could be totally wrong here, I am trying to put myself into their shoes basically.

Reply:
I absolutely agree that they should clearly know which tool belongs to which pack. We could create this many aliases, but I reckon this may become quite unmanageable.

I think we could provide customers with a list along the lines of:

- kong, nginx, coredns, ... -> connectivity
- falco, kyverno, ... -> security
- ...

Which may be some nice common ground.  This also makes it simple to "add/remove" an application, because we just need to update the list (which, for instance, can be in our docs) rather than creating/deleting many aliases.

For instance, in the email template we could add a line with the link to the page with "correspondences" our docs, so that if coredns is broken they just CTRL+F "coredns" and immediately see that that is `connectivity`. Stupid example, but I think it explains what I mean. WDYT?

Reply:
That would work for me as a first iteration